suspect/fitting/singlet.py

```diff
@@ -8,18 +8,32 @@
 
 
 # this is the underlying function for the GaussianPeak model class
-def gaussian(in_data, amplitude, frequency, phase, fwhm):
-    return amplitude * in_data.inherit(suspect.basis.gaussian(in_data.time_axis(), frequency, phase, fwhm))
+def gaussian(time_axis, amplitude, frequency, phase, fwhm):
+    result = amplitude * suspect.basis.gaussian(time_axis, frequency, phase, fwhm)
+    return result
 
 
 # this is the underlying function for the GlobalPhase model class
-def phase_shift(in_data, phase0, phase1):
-    return in_data.spectrum().inherit(np.ones_like(in_data)).adjust_phase(phase0, phase1)
+def phase_shift(in_data, phase0, phase1, spectral_width, num_points):
+    # Update:
+    # Since lmfit updates does not preserve suspect's MRS objects, we
+    # do phase adjustment and spectrum & FID conversion here
+    spectrum = np.fft.fftshift(np.fft.fft(in_data, axis=-1), axes=-1)
+    tmp_data = np.ones_like(in_data)
+    phase_ramp = np.linspace(-spectral_width / 2,
+                             spectral_width / 2,
+                             int(num_points),
+                             endpoint=False)
+    fixed_frequency = 0
+    phase_shift = phase0 + phase1 * (fixed_frequency + phase_ramp)
+    phased_spectrum = tmp_data * np.exp(1j * phase_shift)
```

Comment:
@bennyrowland I found an error below that happens only in MacOS runner. I do not have a Mac machine so it is impossible to test locally what went wrong. 

Based on your email, I assume you are on MacOS — if so, would you mind do some more debugging on this?

For now, I will remove the MacOS runner from CI, but this operation might actually break for MacOS users. 
```
>       phased_spectrum = tmp_data * np.exp(1j * phase_shift)
E       ValueError: operands could not be broadcast together with shapes (1024,) (1023,)
```

Reply:
I found this is due to rounding behavior when converting with `int()`. This is now fixed.